### ISSUE TITLE
[Due for payment 2025-01-30] [$250] Provide education/confirmation before creating workspaces in New Workspace flows

### ISSUE THREAD

**Issue by danielrvidal** [Expensify/App#51504]
Proposal: Provide education/confirmation before creating workspaces in New Workspace flows

Problem: When users attempting to submit expenses are inadvertently led to create workspaces, it causes confusion about how to submit an expense to their boss, leading to frustration and a poor user experience.

Solution: Provide better education about what a workspace is and a confirmation screen when creating them in places employees are running into issues. If a user creates a New Workspace in Global Create or Settings > Workspaces > New Workspace, show a confirmation page informing them what it does and asking them to name it.
- The proposed screens are below.
- We’ll use the same logic we have today to suggest a workspace name, but allow the user to edit it.
- After the user clicks Confirm in the screen, they’ll be dropped into their workspace editor, like they are today.

# Mobile 
![image](https://github.com/user-attachments/assets/7eb8bcab-01f5-463f-828b-a4d4120c0dd2)

# Web
![image](https://github.com/user-attachments/assets/7c4b76f7-a20b-4cc5-b5f2-9d58f9f47541)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021851416153225764502</li>
        <li>Upwork Job ID: 1851416153225764502</li>
        <li>Last Price Increase: 2024-10-30</li>
<li>Automatic offers: </li>
<ul>
<li>allgandalf | Reviewer | 104754886</li>
<li>Krishna2323 | Contributor | 104754887</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @RachCHopkins</details>

**Comment by dubielzyk-expensify** [comment]
Tried to replicate what the request is and it'd probably be something like this:
![image](https://github.com/user-attachments/assets/4c4b18a3-b6ae-43a7-9b02-2ae60daac565)

Though I think I prefer something like this if we wanna list the workspaces with links to them:
![image](https://github.com/user-attachments/assets/6cd8b3c6-90ad-4486-9aa2-4e2413e094b8)

I think ultimately it feels a bit weird to me to mention submitting expenses when they've tapped a New Workspace button, so I kinda wonder if we should be even more generic:
![image](https://github.com/user-attachments/assets/2735c559-a0b4-407b-8828-bb86c4982cb6)

cc @Expensify/design for feedback, musings, poems, et al.

**Comment by shawnborton** [comment]
Yeah, I think I might be with Jon here, how do we know that users are actually hoping to submit an expense when they tap on New Workspace? So that being said, I like the idea of something more generic. 

Here's another riff on that where we use workspace avatars and even link the user to the create expense flow:
![image](https://github.com/user-attachments/assets/9b5846c3-630e-4f2a-b630-6d04d240e9b5)

**Comment by dubielzyk-expensify** [comment]
I quite like that too 😄  I prefer that dialog if we go for the more complex/custom ones.

**Comment by dannymcclain** [comment]
I like Shawn's riff too.

**Comment by danielrvidal** [comment]
We're still working on the design mocks. @dubielzyk-expensify updated the thread with the proposal so we'll likely be ready to work with contributor tomorrow. We'll leave that thread open until tomorrow for any feedback.

Sorry about requesting both designs @dubielzyk-expensify even though we're only going to do the first one right now.

**Comment by dubielzyk-expensify** [comment]
Alright. All good. Let's ignore the modal then 👍

**Comment by danielrvidal** [comment]
@allgandalf just an fyi, I have updated the original post so that the finalized proposal and direction we want to go is there. Please let me know if you have any questions!

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Provide education/confirmation before creating workspaces in New Workspace flows

### What is the root cause of that problem?

This is a new feature.
### What changes do you think we should make in order to solve the problem?
When we create a new WS from the FAB or Workspace list page, we can navigate to a new confirmed workspace page

1. Create a confirmed WS page and add the route and screen for this. The UI follow the design in the description of the OP
2. When we click on new WS button in these case we will navigate to the WS confirm page

https://github.com/Expensify/App/blob/a4c8de60b4cdcd99aad0089ce73c339afa5d94f9/src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.tsx#L467

https://github.com/Expensify/App/blob/a4c8de60b4cdcd99aad0089ce73c339afa5d94f9/src/pages/workspace/WorkspacesListPage.tsx#L366

https://github.com/Expensify/App/blob/a4c8de60b4cdcd99aad0089ce73c339afa5d94f9/src/pages/workspace/WorkspacesListPage.tsx#L399

3. In the WS confirm page, the default value of the WS name input is generated by `generateDefaultWorkspaceName(policyOwnerEmail)` function, the currency is the `localCurrencyCode` of the current user. We can use `CurrencySelectionList` to select another currency.

4. Then when we create a new WS we will call `createWorkspaceWithPolicyDraftAndNavigateToIt` and pass `policyName` as the WS name input in confirm page, we need to accept a new param is `currency` in this function and pass it to `createDraftInitialWorkspace` function and use it [here](https://github.com/Expensify/App/blob/a4c8de60b4cdcd99aad0089ce73c339afa5d94f9/src/libs/actions/Policy/Policy.ts#L1543).

The detail can be done in the PR phrase. I can provide a test branch if we want to test this before moving this forward.
### What alternative solutions did you explore? (Optional)

**Comment by dannymcclain** [comment]
One thing I think we need to consider that [Krishna's proposal](https://github.com/Expensify/App/issues/51504#issuecomment-2445925882) made me realize: I think we should auto-update the avatar when the name field is blurred (or _something_ like that).

It's a bit weird to me that the avatar letter doesn't match the name until the workspace is confirmed. Since this screen is a confirmation, I would expect it to accurately reflect what I'm going to get when I confirm. cc @Expensify/design

**Comment by shawnborton** [comment]
I mean it is probably likely that someone would hit the Save button directly while being focused in the name input... so I wonder if we would even consider updating the avatar after a second of no typing or something even if the user was still focused in the name input?

**Comment by dannymcclain** [comment]
Yeah I was thinking about that too. Seems unlikely they'll actually unfocus the input. Maybe we just update it as soon as the input has a new first character? 🤷

**Comment by shawnborton** [comment]
cc @grgia in case you have any thoughts on this one - how do we determine the color scheme of the avatar too?

**Comment by grgia** [comment]
@shawnborton I believe we use a hash of the policyID to determine the color these days

**Comment by shawnborton** [comment]
Cool, so sounds like the color won't change but we should do something like Danny is suggesting where we take the first letter/etc of the input and adjust the avatar above.

**Comment by shawnborton** [comment]
@Krishna2323 your video looks good, though disclaimer: I am not involved in evaluating proposals :)

**Comment by dannymcclain** [comment]
>disclaimer: I am not involved in evaluating proposals :)

Haha same! But that is what I had in mind.

**Comment by danielrvidal** [comment]
@allgandalf can you review? Does @Krishna2323 looks good since the proposal helped us catch something?

**Comment by danielrvidal** [comment]
Any thoughts on which one to go with @allgandalf ? It would be great to get started!

**Comment by danielrvidal** [comment]
@Krishna2323 @allgandalf I created a second issue because I forgot to also put the confirmation in the workspace chat switcher. 

@allgandalf I assigned you already as C+, but I assume @Krishna2323 can also accept the job and make the updates as they are already doing the work in this flow. Please let me know if that is an issue. 

[[$250] Add workspace creation confirmation to workspace switcher.](https://github.com/Expensify/App/issues/52075#top)

**Comment by rafecolton** [comment]
Realistically it will be a few weeks - @danielrvidal is that soon enough? If not, do you think it would be possible to source a volunteer to take this over?

**Comment by grgia** [comment]
@Krishna2323 @allgandalf could you point me to the BE change request?

**Comment by danielrvidal** [comment]
Thank you for taking a look @grgia, this one is pretty high priority because it's causing a lot of people to be confused.

**Comment by danielrvidal** [comment]
Moving this to Daily until we get an answer from our internal team as this is high priority.

**Comment by grgia** [comment]
> @grgia, the BE endpoint should also accept currency: string; and file: File;. @allgandalf, could you please double-check this as well? Thanks!

@Krishna2323 do we need to return anything in the response?
Also, could you confirm what behavior change we need the API command to make

**Comment by grgia** [comment]
> Not sure what do you mean by "behavior change". Could you please explain?

@Krishna2323 you proposed we send `currency` and `file` to CreateWorkspace-

I want to confirm so we're all on the same page. Are we setting the policy currency and avatar with `currency`/`file`? Then returning that data in the policy response?

**Comment by grgia** [comment]
I've got the Auth PR nearly done- working on unit tests now. I have a draft for the PHP changes that I'm working on as well but haven't gotten to test it yet

**Comment by grgia** [comment]
[Auth PR](https://github.com/Expensify/Auth/pull/13186) is in review, will finalize PHP changes today from the Auth Review

**Comment by grgia** [comment]
[Web PR](https://github.com/Expensify/Web-Expensify/pull/44447) is in final review @allgandalf

**Comment by grgia** [comment]
@allgandalf the changes are in staging 
For the avatar file upload, I referenced the upload avatar flow from ND. I can make an App PR with param updates to help if needed

**Comment by grgia** [comment]
Yep all good to include in the PR!

**Comment by grgia** [comment]
Looking 👀

**Comment by RachCHopkins** [comment]
@allgandalf can you please clarify - is this on prod and ready for payment, or is it still being worked on, and  not ready for payment?

**Comment by shawnborton** [comment]
What's the status on the new PR for this? I would love to get this one into the product ASAP. Thanks!

**Comment by danielrvidal** [comment]
@parasharrajat @allgandalf How is this one going? Is there anything I can do to help get this one done?

**Comment by RachCHopkins** [comment]
@allgandalf can you please do the checklist etc?

**Comment by RachCHopkins** [comment]
Payment Summary:

- Contributor: @Krishna2323 to be paid $250 via Upwork
- Contributor+: @allgandalf to be paid $250 via Upwork

Upwork job [here](https://www.upwork.com/jobs/~021851416153225764502)

**Comment by RachCHopkins** [comment]
Contributor has been paid, the contract has been completed, and the Upwork post has been closed.
